### PR TITLE
Feat: NonRESTful Merchant FIND

### PR DESCRIPTION
**Pull request by GBowman1**
## Type of change 
- [x] Feature
- [ ] Debug
- [ ] Refactor

## Describe Changes Made
Created NonRESTful endpoint for merchant name search, created controller action in Merchant controller and a look up method in Merchant model.

- I realized my solution to get merchant data from item ID isnt restful since using merchant controller action should probably update routes to send to a new controller and make an action for it.

- EVERYTHING ive commited still needs SAD path and edge case testing

## PR Checklist
- [x] Added Reviewer
- [x] Followed TDD, And Test are Passing
- [ ] Linked Issues for Project Board

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 class Merchant < ApplicationRecord
     has_many :items
+
+    def self.find_merchant_by_name(name)
+        where('lower(name) LIKE ?', "%#{name.downcase}%").order('lower(name)').first
```
Comment (LJ9332) [inline app/models/merchant.rb:5]: Great job with this method!

**Review by LJ9332 — APPROVED** [review]
